src/secret_manager/infisical.ts

```diff
@@ -0,0 +1,111 @@
+        }
+        if (!process.env.SECRET_MANAGER_INFISICAL_ENV) {
+            throw new Error("SECRET_MANAGER_INFISICAL_ENV environment variable is required");
+        }
+
+        super();
+        this.clientId = process.env.SECRET_MANAGER_INFISICAL_CLIENT_ID || "";
+        this.clientSecret = process.env.SECRET_MANAGER_INFISICAL_CLIENT_SECRET || "";
+        this.projectId = process.env.SECRET_MANAGER_INFISICAL_PROJECT_ID || "";
+        this.env = process.env.SECRET_MANAGER_INFISICAL_ENV || "";
+        this.client = new InfisicalSDK({
+            siteUrl: process.env.SECRET_MANAGER_INFISICAL_API_URI
+          });
+        this.connect();
+    }
+
+    async connect(): Promise<void> {
+        try {
+            // Authenticate with Infisical
+            await this.client.auth().universalAuth.login({
+                clientId: this.clientId,
+                clientSecret: this.clientSecret
+            });
+            console.log("Connected successfully to iNFISICAL");
+        }
+        catch (err) {
+            console.error("Connection to Bitwarden failed", err);
+        }
+    }
+
+    // SECRETS
+
+    async addSecret(key: string, params: any): Promise<string> {
+        //JSON param before sending
+        const stringParams: string = JSON.stringify(params);        
+        var createdSecret = await this.client.secrets().createSecret(key, {
+            projectId: this.projectId,
+            secretValue: stringParams,
+            environment: this.env,
+            secretPath:  "/",
+        });
+        return createdSecret.approval.id;
+    }
+
+    async getSecret(id: string): Promise<any | null> {
+        try {
+            const secret = await this.client.secrets().getSecret({
+                projectId: this.projectId,
+                secretName: id,
+                environment: this.env,
+                secretPath: "/",
+              });
+            return JSON.parse(secret.secretValue);
+        }
+        catch (err) {
+            if (err instanceof Error && err.message.includes("[404 Not Found]")) {
+                return null;
+            }
+            throw err;
+        }
+    }
+
+    async deleteSecret(id: string): Promise<void> {
+        await this.deleteSecrets([id]);
+    }
+
+    async deleteSecrets(ids: string[]): Promise<void> {
+        try {
+            for (const secretName of ids) {
+                try {
+                  await this.client.secrets().deleteSecret(secretName, {
+                    projectId: this.projectId,
+                    environment: this.env,
+                    secretPath: "/"
+                  });
+                  console.log(`Secret "${secretName}" deleted.`);
+                } catch (error) {
+                  console.error(`Error deleting "${secretName}":`, error);
+                }
+              }
+        }
+        catch (err) {
+            if (!(err instanceof Error && err.message.includes("[404 Not Found]"))) {
```

Comment:
In Bitwarden implementation, this error handling is for when the credential is in the mongodb but not in the secret manager.

Error handling here might be different from Bitwarden.